lib/streamlit/app_session.py

```diff
@@ -512,7 +514,8 @@ def _create_new_session_message(self) -> ForwardMsg:
         """Create and return a new_session ForwardMsg."""
         msg = ForwardMsg()
 
-        msg.new_session.script_run_id = _generate_scriptrun_id()
+        self._script_run_id = _generate_scriptrun_id()
+        msg.new_session.script_run_id = self._script_run_id
```

Comment:
And then this function should probably take the script_run_id as a param, rather than generating it anew:

```python
def _create_new_session_message(self, script_run_id: str) -> ForwardMsg:
  ...
```